### PR TITLE
Fix unexpected error in chat

### LINKED ISSUE #830: Bug: "Application error: a client-side exception has occured" when using messages
**Describe the bug 1**
A: When I am in a chat with another user at climateconnect.earth and then log out from there I get the error: "Application error: a client-side exception has occurred".
B: When I log out from my inbox a loading graphic (Climate Connect logo + text "loading ...") is shown forever. Refreshing the page gives me a 404 error in the browser window and a failed get request in the console (error 500).

**To Reproduce**
Steps to reproduce the behavior A:
1. Open a chat with another user on climateconnect.earth
2. Log out

And B:
1. Open your inbox on climateconnect.earth
2. Log out

**Describe the bug 2**
This seems to happen randomly, about 50% of the time. When sending a message to another user on climateconnect.earth the message is first sent and displayed correctly but then the "Application error: a client-side exception has occurred" turns up. After refreshing, everything seems normal.

**To Reproduce**
1. Send a message to another user on climateconnect.earth
2. You might need to repeat since the error does not happen every time

Not sure whether bugs 1 and 2 are related.

### PR DESCRIPTION
**Pull request by positiveimpact**
## Description

- Fixed index out of bound error in backend which through a 500 error. 
- For some reason the websocket sends each chat message twice. Fixed frontend implications of that by only showing a message once per id. 
- Fixed the unexpected error when logging out while in the chat window. This was caused by trying to access a property of the user object which is obv. null if you log out.

Closes #830 

## Test plan

1. Open a private chat (ideally with 2 accounts in 2 browsers)
2. Write a message
3. Make sure there is no error on either browser and the message arrives exactly once. 

Fixes were made to websocket mode, so you shouldn't be using the fallback HTTP requests

### CODE REVIEW

File: backend/chat_messages/utility/chat_setup.py [hunk 1]
```diff
@@ -26,15 +26,17 @@ def set_read(messages, user, is_private_message):
...
+            unread_user_notification = UserNotification.objects.get(
+                notification__in=message_notifications,
+                read_at=None,
+                user=user
+            )
+            user_notification = unread_user_notification
             user_notification.read_at = datetime.now()
             user_notification.save()
+        except UserNotification.DoesNotExist:
+            print("there is no user notification for "+user.first_name)
```
Comment (samschaevitz) [inline backend/chat_messages/utility/chat_setup.py:39]: do you want to leave this as print() vs a logging statement?

File: frontend/src/components/layouts/layout.js [hunk 2]
```diff
@@ -16,11 +16,11 @@ const useStyles = makeStyles((theme) => ({
     textAlign: "center",
     margin: `${theme.spacing(4)}px 0`,
   },
-  alert: {
+  alert: props => ({
     width: "100%",
     zIndex: 100,
-    marginTop: theme.spacing(-2),
-  },
+    marginTop: !props.donationCampaignRunning && theme.spacing(-2),
```
Comment (samschaevitz) [inline frontend/src/components/layouts/layout.js:22]: Is this related to the error checking above ? If not maybe split into another PR...

**Review by samschaevitz — APPROVED** [review]
Looks mostly fine, PTAL before merging

File: frontend/src/components/layouts/layout.js (hunk 2 above)
Comment (positiveimpact) [reply frontend/src/components/layouts/layout.js:22]: I think I would leave it in here because this is another bug that happens when logging out while in the chat window: You'll get redirected to /signin and the page will look weird because the donation banner goes over the message telling you that you need to log in to chat

File: backend/chat_messages/utility/chat_setup.py (hunk 1 above)
Comment (positiveimpact) [reply backend/chat_messages/utility/chat_setup.py:39]: Changed it to logging, thanks for pointing that out
